### PR TITLE
Fixes #99: Using standard lightblue::eap::client resource for metadata-mgmt app

### PR DESCRIPTION
**Pull request by paterczm**
There are 2 changes here:
* Creating module paths using mkdir -p, because puppet's way of recursive directory creation is doomed to generate conflicts
* Using lightblue::eap::client resource to configure lightblue client for metdata-mgmt app. This is the same resource all other apps are using.

The real driver for this change is https://github.com/lightblue-platform/lightblue-applications/issues/99

### CODE REVIEW

File: manifests/eap/client/modulepath.pp [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
+    $module_path = '/usr/share/jbossas/modules/com/redhat/lightblue/client'
 
-    file { $module_base :
-        ensure   => 'directory',
-        owner    => 'jboss',
-        group    => 'jboss',
-        mode     => '0755',
+    exec { $module_path:
+        command => "mkdir -p $module_path",
+        user    => 'jboss',
```
Comment (jewzaam) [inline manifests/eap/client/modulepath.pp]: Directory permissions (mode) need set.  We had 0755 before but this isn't appropriate.  Only root and the jboss user should read contents of this directory.

File: manifests/eap/client/modulepath.pp [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
+    #    '/usr/share/jbossas/modules/com/redhat/lightblue',
+    #    '/usr/share/jbossas/modules/com/redhat/lightblue/client']
+
+    #file { $module_base :
+    #    ensure   => 'directory',
+    #    owner    => 'jboss',
+    #    group    => 'jboss',
+    #    mode     => '0755',
+    #}
+
```
Comment (jewzaam) [inline manifests/eap/client/modulepath.pp]: Why add commented out code?

**Comment by paterczm** [comment]
> Why add commented out code?

I left it to make it clear what the standard puppet solution is and why it's not a good fit. If it makes things less clear, then it's against my intentions.

File: manifests/eap/client/modulepath.pp (hunk 1 above)
Comment (paterczm) [inline manifests/eap/client/modulepath.pp]: You're right. Certificates and passwords should not be readable to anyone, like other jboss modules. Pushed a fix.

Comment (dcrissman) [inline manifests/eap/client/modulepath.pp]: I think you want to keep the group => 'jboss', just set the mode to something like 700 or 600.

Comment (paterczm) [inline manifests/eap/client/modulepath.pp]: I'm running the command as jboss user, I don't think I need to specify the group as well. Set mode to 440 (r--r-----).

File: manifests/client/configure.pp [hunk 3]
```diff
@@ -43,7 +43,7 @@
 
     file { $title:
         ensure => 'file',
-        mode    => '0644',
+        mode    => '0550',
```
Comment (jewzaam) [inline manifests/client/configure.pp]: Couldn't this be 0440?  There's not need for execute on the config file.
This applies to most of the other permissions changed in this PR, not going to annotate them all.

File: manifests/eap/module.pp [hunk 4]
```diff
@@ -195,24 +195,6 @@
...
-
-    file { '/usr/share/jbossas/modules/com/redhat/lightblue/main/lightblue-client.properties':
-        mode    => '0644',
-        owner   => 'jboss',
-        group   => 'jboss',
-        content => template('lightblue/properties/appconfig.properties.erb'),
-        notify  => Service['jbossas'],
-        require => File[$directory],
-    }
-
```
Comment (jewzaam) [inline manifests/eap/module.pp]: Should we puppetize cleanup (removal) of the appconfig.properties and lightblue-client.properties files?  Else they hang out on systems forever.

**Pull request merged**